cmd/wgcf-cli/cmd_generate.go

```diff
@@ -13,64 +16,204 @@ import (
 
 var generateCmd = &cobra.Command{
 	Use:       "generate",
-	Short:     "Generate a xray/sing-box config",
+	Short:     "Generate a xray/sing-box/wg-quick config",
 	Run:       generate,
 	Args:      cobra.OnlyValidArgs,
-	ValidArgs: []string{"--xray", "--sing-box", "--wg"},
+	ValidArgs: []string{"--xray", "--xray-module", "--xray-tag", "--xray-indent-width", "--sing-box", "--wg", "--wg-quick", "--output-file"},
+}
+
+type OutputFileType int8
+
+const (
+	Stdout OutputFileType = iota
+	Default
+	Custom
+)
+
+type GeneratorType int8
+
+const (
+	Xray GeneratorType = iota
+	SingBox
+	WgQuick
+	None
+)
+
+func (t GeneratorType) String() string {
+	switch t {
+	case Xray:
+		return "xray"
+	case SingBox:
+		return "sing-box"
+	case WgQuick:
+		return "wg-quick"
+	}
+	return "unknown"
 }
 
 func init() {
 	rootCmd.AddCommand(generateCmd)
-	generateCmd.Flags().Bool("xray", false, "generate a xray config")
-	generateCmd.Flags().Bool("sing-box", false, "generate a sing-box config")
-	generateCmd.Flags().Bool("wg", false, "generate a wg-quick config")
+	generateCmd.Flags().Bool(asString(Xray), false, "generate a xray config")
+	generateCmd.Flags().Bool(asString(SingBox), false, "generate a sing-box config")
+	generateCmd.Flags().Bool(asString(WgQuick), false, "generate a wg-quick config")
+	generateCmd.Flags().Bool("wg", false, "see --"+asString(WgQuick))
+
+	generateCmd.Flags().String("output-file", "default", "output file name. Supported values: 'default'/'stdout'/any file path")
+	generateCmd.Flags().String(asString(Xray)+"-module", "", "xray top-level config module ('inbounds' as example). By default generate no top-level module")
+	generateCmd.Flags().String(asString(Xray)+"-tag", "wireguard", "'Tag' field of xray config")
+	generateCmd.Flags().Uint8(asString(Xray)+"-indent-width", 4, "indentation size for xray config")
 }
 
-func generate(cmd *cobra.Command, args []string) {
+func asString[V fmt.Stringer](object V) string {
+	return V.String(object)
+}
+
+func detectOutputFileType(cmd *cobra.Command) (OutputFileType, error) {
 	var err error
-	xray, _ := cmd.Flags().GetBool("xray")
-	sing, _ := cmd.Flags().GetBool("sing-box")
-	wg, _ := cmd.Flags().GetBool("wg")
+	path, err := cmd.Flags().GetString("output-file")
+	if err != nil {
+		return Stdout, err
+	}
+	switch path {
+	case "stdout":
+		return Stdout, nil
+	case "default":
+		return Default, nil
+	}
+	return Custom, nil
+}
 
-	var resStruct C.Response
-	body := utils.ReadConfig(configPath)
-	if err := json.Unmarshal(body, &resStruct); err != nil {
-		fmt.Fprintln(os.Stderr, "Error:", err)
-		os.Exit(1)
+func ternary[V any](condition bool, on_true V, on_false V) V {
+	if condition {
+		return on_true
+	}
+	return on_false
+}
```

Comment:
```golang
func oneTrue(bools ...bool) bool {
    f := false
    for _, b := range bools {
        if b {
            if f {
                return false
            }
            f = true
        }
    }
    return f
}
```
I think using this type of function is better. We can also adjust it to suit our condition where `c == 0`.

Reply:
Hmm, I don't thought about this way. This is definitely better solution. I will update this pull request

Reply:
See [1e9ab18](https://github.com/ArchiveNetwork/wgcf-cli/pull/34/commits/1e9ab1886f40fdef10d49b9719336b44d32ae5ec)